modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-volley/request/IRequestFactory.mustache

```diff
@@ -0,0 +1,78 @@
+package {{packageName}}.request
+
+import com.android.volley.Request
+import com.android.volley.Response
+import java.io.UnsupportedEncodingException
+import java.lang.reflect.Type
+import java.net.URLEncoder
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+interface IRequestFactory {
+
+    companion object {
+        /**
+         * ISO 8601 date time format.
+         * @see https://en.wikipedia.org/wiki/ISO_8601
+         */
+        val DATE_TIME_FORMAT = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
```

Comment:
@alisters We should use OffsetDateTime/LocalDate instead of Date and DateTimeFormatter instead of SimpleDateFormat everywhere. Or do you have a requirement to use Date?

Reply:
Changed to OffsetDateTime/LocalDate and DateTimeFormatter. Some additional points
- The formats used by the DateTimeFormatter are hardcoded. We lifted the previous code from a different open api library I think but don't use this code - in our api date's are represented as strings without formats. I believe this may have been due to incompatible code gen in different libraries that we use when a format was applied. 
- Petstore has a sample api with a string of format "Date Time" but not one of just "date"
TL;DR - i think the above code is unexercised and hence maybe untested, but agree with the change, and if this were done in all libraries maybe we could annotate our date times with formats.

Reply:
According to the openapi spec, here are the date formats that should be used.

- `date` – full-date notation as defined by RFC 3339, section 5.6, for example, `2017-07-21`
- `date-time` – the date-time notation as defined by RFC 3339, section 5.6, for example, `2017-07-21T17:32:28Z`

https://swagger.io/docs/specification/data-models/data-types/

Reply:
In the modified code i use 
        fun formatDateTime(datetime: OffsetDateTime) = DateTimeFormatter.ISO_INSTANT.format(datetime)
        fun formatDate(date: LocalDate) = DateTimeFormatter.ISO_LOCAL_DATE.format(date)

I think i have these correct as https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html shows them as lining up with the swagger data-types/RFC references